### PR TITLE
Complete repository refactoring - method renaming

### PR DESCRIPTION
**Pull request by kaizimmerm**
Last step of https://github.com/eclipse/hawkbit/issues/197

- Renamed all methods according to the defined schema
- Split TagManagement
- In addition a few cleanups here and there

Reviewers: @Jonkno @michahirsch

### CODE REVIEW

**Comment by michahirsch** [comment]
like the new method-naming convention 👍

File: hawkbit-repository/hawkbit-repository-api/src/main/java/org/eclipse/hawkbit/repository/DistributionSetTypeManagement.java [hunk 1]
```diff
@@ -106,74 +38,24 @@
...
-     *             if the {@link DistributionSetType} is already in use by a
-     *             {@link DistributionSet} and user tries to change list of
-     *             {@link SoftwareModuleType}s
-     * @throws ConstraintViolationException
-     *             if fields are not filled as specified. Check
-     *             {@link DistributionSetTypeUpdate} for field constraints.
-     */
-    @PreAuthorize(SpringEvalExpressions.HAS_AUTH_UPDATE_REPOSITORY)
-    DistributionSetType updateDistributionSetType(@NotNull DistributionSetTypeUpdate update);
+    Optional<DistributionSetType> get(@NotNull Long id);
```
Comment (Jonkno) [inline hawkbit-repository/hawkbit-repository-api/src/main/java/org/eclipse/hawkbit/repository/DistributionSetTypeManagement.java]: Use same name for id parameter within the class -> for example distributionSetTypeId

File: hawkbit-repository/hawkbit-repository-api/src/main/java/org/eclipse/hawkbit/repository/SoftwareModuleManagement.java [hunk 2]
```diff
@@ -258,34 +169,35 @@
      *             is module with given ID does not exist
      */
     @PreAuthorize(SpringEvalExpressions.HAS_AUTH_READ_REPOSITORY)
-    Optional<SoftwareModuleMetadata> findSoftwareModuleMetadata(@NotNull Long moduleId, @NotEmpty String key);
+    Optional<SoftwareModuleMetadata> getMetaDataBySoftwareModuleId(@NotNull Long moduleId, @NotEmpty String key);
```
Comment (Jonkno) [inline hawkbit-repository/hawkbit-repository-api/src/main/java/org/eclipse/hawkbit/repository/SoftwareModuleManagement.java]: Use the same name for softwareModule id in all method parameter

File: hawkbit-repository/hawkbit-repository-api/src/main/java/org/eclipse/hawkbit/repository/SoftwareModuleManagement.java [hunk 3]
```diff
@@ -183,7 +117,7 @@
      *             if distribution set with given ID does not exist
      */
     @PreAuthorize(SpringEvalExpressions.HAS_AUTH_READ_REPOSITORY)
-    Page<SoftwareModule> findSoftwareModuleByAssignedTo(@NotNull Pageable pageable, @NotNull Long setId);
+    Page<SoftwareModule> findByAssignedTo(@NotNull Pageable pageable, @NotNull Long setId);
```
[the remark this answers is not in the document]
Comment (kaizimmerm) [reply hawkbit-repository/hawkbit-repository-api/src/main/java/org/eclipse/hawkbit/repository/SoftwareModuleManagement.java]: No, the field is called "assignedTo"

File: hawkbit-repository/hawkbit-repository-api/src/main/java/org/eclipse/hawkbit/repository/SoftwareModuleManagement.java (hunk 2 above)
Comment (kaizimmerm) [reply hawkbit-repository/hawkbit-repository-api/src/main/java/org/eclipse/hawkbit/repository/SoftwareModuleManagement.java]: Its now moduleId

**Pull request merged**